src/NServiceBus.Core.Tests/Sagas/ComponentTests/SagaPersisterTests.cs

```diff
@@ -0,0 +1,208 @@
+        protected Task<TSagaData> GetByIdAndUpdate(Guid sagaId, Action<TSagaData> update, params Type[] availableTypes) => GetByIdAndUpdate<TSaga, TSagaData>(sagaId, update, availableTypes);
+        protected Task<TSagaData> GetByCorrelationPropertyAndUpdate(string correlatedPropertyName, object correlationPropertyData, Action<TSagaData> update) => GetByCorrelationPropertyAndUpdate<TSaga, TSagaData>(correlatedPropertyName, correlationPropertyData, update);
+        protected Task<TSagaData> GetByCorrelationPropertyAndComplete(string correlatedPropertyName, object correlationPropertyData) => GetByCorrelationPropertyAndComplete<TSaga, TSagaData>(correlatedPropertyName, correlationPropertyData);
+        protected Task<TSagaData> GetByCorrelationProperty(string correlatedPropertyName, object correlationPropertyData) => GetByCorrelationProperty<TSaga, TSagaData>(correlatedPropertyName, correlationPropertyData);
+        protected Task<TSagaData> GetById(Guid sagaId, params Type[] availableTypes) => GetById<TSaga, TSagaData>(sagaId, availableTypes);
+    }
+
+    public class SagaPersisterTests
+    {
+        protected PersistenceTestsConfiguration configuration;
+
+        [OneTimeSetUp]
+        public async Task OneTimeSetUp()
+        {
+            configuration = new PersistenceTestsConfiguration();
+            await configuration.Configure();
+        }
+
+        [OneTimeTearDown]
+        public async Task OneTimeTearDown()
+        {
+            await configuration.Cleanup();
+        }
+
+        protected async Task SaveSaga<TSaga, TSagaData>(TSagaData saga, params Type[] availableTypes)
+            where TSaga : Saga<TSagaData>, new()
+            where TSagaData : IContainSagaData, new()
+        {
+            var insertContextBag = configuration.GetContextBagForSagaStorage();
+            using (var insertSession = await configuration.SynchronizedStorage.OpenSession(insertContextBag))
+            {
+                var correlationProperty = SetActiveSagaInstanceForSave(insertContextBag, new TSaga(), saga, availableTypes);
+
+                await configuration.SagaStorage.Save(saga, correlationProperty, insertSession, insertContextBag);
+                await insertSession.CompleteAsync();
+            }
+        }
+
+        protected async Task<TSagaData> GetByIdAndComplete<TSaga, TSagaData>(Guid sagaId, params Type[] availableTypes)
+            where TSaga : Saga<TSagaData>, new()
+            where TSagaData : IContainSagaData, new()
+        {
+            var context = configuration.GetContextBagForSagaStorage();
+            TSagaData sagaData;
+            var persister = configuration.SagaStorage;
+            using (var completeSession = await configuration.SynchronizedStorage.OpenSession(context))
+            {
+                SetActiveSagaInstanceForGet<TSaga, TSagaData>(context, new TSagaData(), availableTypes);
```

Comment:
It was primarily added because SqlPersistence was assuming that the ActiveSagaInstance class is in the context bag that is passed to the methods. The core adds it. We wanted to stay true to that simulation that's why we added it and therefore that's why there is this noise. Now that SqlPersistence doesn't require it anymore on latest develop we could essentially remove this "noise" like you called it but then we would also remove one simulation piece of the puzzle. We still haven't decided who is the source of truth for these tests therefore I'd say we leave it true to how the tests were originally defined.